client/src/featureform/serving.py

```diff
@@ -1006,6 +1010,17 @@ def dataframe(self) -> pd.DataFrame:
         """
         if self._dataframe is not None:
             return self._dataframe
+        elif spark_session is not None:
+            req = serving_pb2.TrainingDataRequest()
+            req.id.name = self._stream.name
+            req.id.version = self._stream.version
+            resp = self._stream._stub.ResourceLocation(req)
+
+            file_format, location = self.get_file_format_and_location(resp.location)
+            self._dataframe = self.get_spark_dataframe(
```

Comment:
do you have a python file that shows what this'll look like? Is there a way to do this so you don't have to pass in a spark session?

Reply:
```
df = client.training_set("name", "variant").dataframe(spark_session=spark)
```
Unfortunately, you can't because the spark session sets the credentials needed to access the file path.

Reply:
but don't we have all of that stored on the host?

Reply:
we don't want to retrieve credentials from host. It can lead to more problems.